3 add content pages to flet web app

davidbuu opened the pull request:
Push request 1: Hey Chris, segne das mal ab!

christophschaller committed:
feat: add basic web app frame

christophschaller committed:
feat: add basic web app frame

christophschaller committed:
feat: add basic web app frame

christophschaller committed:
feat: add basic web app frame

christophschaller committed:
fix: return ids and objects instead of half empty tuples

christophschaller committed:
refactor: remove pages build on ft.UserControl

christophschaller committed:
refactor: remove pages build on ft.UserControl

christophschaller committed:
refactor: remove pages build on ft.UserControl

christophschaller committed:
refactor: remove pages build on ft.UserControl

christophschaller committed:
refactor: remove app_layout.py

christophschaller committed:
refactor: remove app_layout.py

christophschaller committed:
refactor: remove app_layout.py

christophschaller committed:
refactor: remove app_layout.py

christophschaller committed:
feat: add views, all inheriting from BaseView / ft.View

christophschaller committed:
refactor: make use of new views & refactor for readability

christophschaller committed:
refactor: make use of new views & refactor for readability

christophschaller committed:
refactor: make use of new views & refactor for readability

christophschaller committed:
refactor: make use of new views & refactor for readability

davidbuu committed:
Update main.py

davidbuu committed:
get_user_activities und Tabelle eingefügt

get_user_activities(self, user_id: str, sport_type: str=None, start_date: datetime=None, end_date: datetime=None)

davidbuu committed:
get_user_activities und Tabelle eingefügt

get_user_activities(self, user_id: str, sport_type: str=None, start_date: datetime=None, end_date: datetime=None)

davidbuu committed:
get_user_activities und Tabelle eingefügt

get_user_activities(self, user_id: str, sport_type: str=None, start_date: datetime=None, end_date: datetime=None)

davidbuu committed:
get_user_activities und Tabelle eingefügt

get_user_activities(self, user_id: str, sport_type: str=None, start_date: datetime=None, end_date: datetime=None)

davidbuu committed:
black, docsting

davidbuu committed:
black, docsting

davidbuu committed:
black, docsting

davidbuu committed:
challenge class, tabelle und plot

davidbuu committed:
challenge class, tabelle und plot

davidbuu committed:
challenge class, tabelle und plot

davidbuu committed:
chart fixes

davidbuu committed:
chart fixes

davidbuu committed:
chart fixes

davidbuu committed:
chart fixes

davidbuu committed:
Run fix

davidbuu committed:
Run fix

davidbuu committed:
Run fix

davidbuu committed:
Diagramm kann jetzt auch auf Wochenbasis umgestellt werden

davidbuu committed:
Diagramm kann jetzt auch auf Wochenbasis umgestellt werden

davidbuu committed:
Diagramm kann jetzt auch auf Wochenbasis umgestellt werden

davidbuu committed:
User_View sport Table

davidbuu committed:
User_View sport Table

davidbuu committed:
User_View sport Table

davidbuu committed:
add activity plot on user screen

davidbuu committed:
add activity plot on user screen

davidbuu committed:
add activity plot on user screen

davidbuu committed:
Merge branch 'main' into 3-add-content-pages-to-flet-web-app

christophschaller committed:
Merge remote-tracking branch 'origin/3-add-content-pages-to-flet-web-app' into 3-add-content-pages-to-flet-web-app

# Conflicts:
#	metriker_app/metriker_app/main.py
#	metriker_app/metriker_app/views/__init__.py
#	metriker_app/metriker_app/views/base_view.py
#	metriker_app/metriker_app/views/challenges_view.py
#	metriker_app/metriker_app/views/data_privacy_view.py
#	metriker_app/metriker_app/views/login_view.py
#	metriker_app/metriker_app/views/user_view.py

christophschaller committed:
feat: make long challenges scrollable

christophschaller committed:
feat: make long challenges scrollable

christophschaller committed:
lint: update root dependencies

christophschaller committed:
lint: update root dependencies

christophschaller commented on metriker_app/metriker_app/views/challenges_view.py:
```diff
@@ -0,0 +1,364 @@
+                    # if activity.type == sport
+                ]
+            )
+            print(distance)
```
No print statements in final code

christophschaller committed:
fix: remove padding interfering with buttons

christophschaller committed:
refactor: remove __init__.py as dir is not a package

christophschaller committed:
refactor: remove __init__.py as dir is not a package

christophschaller committed:
chore: make py version more specific <4

christophschaller committed:
Merge remote-tracking branch 'origin/3-add-content-pages-to-flet-web-app' into 3-add-content-pages-to-flet-web-app

# Conflicts:
#	database_utils/database_utils/activity_handler.py
#	metriker_app/metriker_app/views/challenges_view.py
#	metriker_app/metriker_app/views/user_view.py
#	poetry.lock
#	pyproject.toml

christophschaller committed:
fix: activity.type to activity.sport_type

davidbuu committed:
ruff konform

merkert noch wegen timezone und print befehlen (evtl zu logging umwandeln)

davidbuu committed:
ruff ist zufrieden

davidbuu committed:
ruff sagt go und geblacked ist es auch

davidbuu committed:
letzt ruff fehler in activity handler begradigt

Docstring fehler

davidbuu commented on metriker_app/metriker_app/views/challenges_view.py:
```diff
@@ -0,0 +1,364 @@
+}
+
+
+class ChallengeContent(ft.Column):
```
Noch offen

davidbuu committed:
datetime änderung

davidbuu committed:
ChallengeContent in eigenes Modul ausgelagert

davidbuu committed:
change request: remove unused challenges

@dataclass
class Challenge:
    """Temporary Dataclass to define a challenge."""

    name: str
    icon: str
    content: ft.Control

CHALLENGES = {
    "bike": Challenge(
        name="bike",
        icon=ft.icons.PEDAL_BIKE,
        content=ft.Container(content=ft.Text("Bike")),
    ),
    "run": Challenge(
        name="run",
        icon=ft.icons.HIKING,
        content=ft.Container(content=ft.Text("run")),
    ),
}

davidbuu commented:
Alle change requests sind eingebaut.

davidbuu commented:
Muss ich den Pull request wieder anstossen? Wenn ja wie? Finde keinen Button der dazu einlädt?
